bitsv/network/services/bchsvexplorer.py

```diff
@@ -0,0 +1,105 @@
+import requests
+import json
+from decimal import Decimal
+
+from bitsv.network import currency_to_satoshi
+from bitsv.network.meta import Unspent
+
+# left here as a reminder to normalize get_transaction()
+from bitsv.network.transaction import Transaction, TxInput, TxOutput
+from bitsv.constants import BSV
+
+DEFAULT_TIMEOUT = 30
+BSV_TO_SAT_MULTIPLIER = BSV
+
+
+class BchSVExplorerDotComAPI:
+    """
+    Simple bitcoin SV REST API --> uses base58 address format (addresses start with "1")
+    - get_address_info
+    - get_balance
+    - get_transactions
+    - get_transaction
+    - get_unspent
+    - broadcast_tx
+    """
+    MAIN_ENDPOINT = 'https://bchsvexplorer.com/'
```

Comment:
I would have to see what you mean exactly. I think over-classing something can make it a bit more complicated and harder to override. I would consider having a endpoint=MAIN_ENDPOINT on every function but I think this is at least an interim step worth releasing, or close to it.

Reply:
I was just thinking it could be something like this:
```
class BaseServiceAPI:
        @classmethod
        def get_address_info(cls, address):
            r = requests.get(cls.ADDRESS_INFO_ENDPOINT.format(address), timeout=DEFAULT_TIMEOUT)
            r.raise_for_status()  # pragma: no cover
            return r.json()

class API1(BaseServiceAPI):
    ADDRESS_INFO_ENDPOINT = 'api/addr/{}'

class API2(BaseServiceAPI):
    ADDRESS_INFO_ENDPOINT = 'address/{}/info'

Reply:
Interesting. I honestly don't use classes hardly at all. I have also used generic functions, like say:

```def json_request(api_call):
    r = requests.get(ENDPOINT + api_call, timeout=DEFAULT_TIMEOUT)
    r.raise_for_status()
    return r.json

def get_transactions(address):
    return json_request('/api/addr/{}'.format(address))
```

I would feel more comfortable working with that style as I know better what's going on. I'm certain classes would work, just not what I'm used to.

Do you think it's better to merge as-is and plan on that later, or to try to shorten the code this way before the next release?

Reply:
Ok, looks like you approved it for now :).

Reply:
I think it works for now, although it might not be the most pythonic architecture. The important thing is that the codebase is readable so that we can maintain it without introducing additional issues 👍